extensions/nextdns/src/components/add-domain.tsx

```diff
@@ -0,0 +1,63 @@
+import { Action, ActionPanel, Form, Icon, showToast, Toast, useNavigation } from "@raycast/api";
+import { useForm } from "@raycast/utils";
+import isValidDomain from "is-valid-domain";
+import { DomainSubmitValues, Mutate } from "../types";
+import { addDomain } from "../libs/api";
+
+export default function AddDomain(props: { type: string; mutate?: Mutate | undefined }) {
+  const { pop } = useNavigation();
+  const { type, mutate } = props;
+  const description =
+    type === "allow"
+      ? "Allowing a domain will automatically allow all its subdomains. Allowing takes precedence over everything else, including security features."
+      : "Denying a domain will automatically deny all its subdomains.";
+  const { itemProps, handleSubmit } = useForm<DomainSubmitValues>({
+    async onSubmit(values) {
+      if (mutate) {
```

Comment:
Is there a use-case where `undefined` is being sent for `mutate`? In this diff, at least mutate is populated always. I think we should simplify the logic here for easier maintenance down the line.

Reply:
Yes, from `empty-view.tsx`

```ts
{mutate && (
   <Action.Push
     icon={Icon.Plus}
     title="Add New Domain"
     shortcut={Keyboard.Shortcut.Common.New}
     target={<AddDomain type={type ?? ""} mutate={mutate ?? undefined} />}
   />
)}
```

Reply:
Ah ok, but anyways what I was pointing to was maybe we don't render an `Action.Push` in cases where `mutate` is `undefined`, rather than rendering a form and having an edge-case where submitting can be a no-op as the current implementation suggests. That way we can always expect a `mutate` value for Form.

Reply:
Makes sense 🙂